File: 14-longest-common-prefix/longest-common-prefix.ts
function longestCommonPrefix(strs: string[]): string {
    let prefix = strs[0]

    for (let i = 1; i < strs.length; i++) {
        const currentString = strs[i]

        while (currentString.indexOf(prefix) !== 0) {
            prefix = prefix.substring(0, prefix.length - 1)
            if (prefix === '') return ''
        }
    }

    return prefix
};

File: 14-longest-common-prefix/longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: list[str]) -> str:
        if not strs: 
            return ""
        
        for i in range(len(strs[0])):
            char = strs[0][i]
            for s in strs[1:]:
                if i == len(s) or s[i] != char:
                    return strs[0][:i]

        return strs[0]